### PR TITLE
Add microsoft case study

### PR DESCRIPTION
**Pull request by clearlyTHUYDOAN**
## Description

Adds Microsoft case study and related assets.

### CODE REVIEW

File: src/content/caseStudies/microsoft/index.mdx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  title: "Web Pages in Half the Time: Why Microsoft Chose Astro to Build Their Fluent 2 Design System Website"
+  description: "How Microsoft, a household name, made large performance gains using Astro for their Fluent 2 design system website."
+coverGradientFrom: "#3245FF"
+coverGradientTo: "#BC52EE"
+publishDate: "September 18, 2023"
+authors:
+  - astro-team
+lang: "en"
+socialImage: /src/content/blog/_images/microsoft/social-image.webp
+coverImage: /src/content/blog/_images/firebase/cover-image.webp
```
Comment (clearlyTHUYDOAN) [inline src/content/caseStudies/microsoft/index.mdx:14]: Replace this with Microsoft cover image.

File: src/content/caseStudies/microsoft/index.mdx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+headerImage: /src/content/blog/_images/firebase/header.webp
```
Comment (clearlyTHUYDOAN) [inline src/content/caseStudies/microsoft/index.mdx:15]: Replace this with transparent Microsoft header image.

File: src/content/caseStudies/microsoft/index.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+The team wanted something lightweight and low maintenance on the DevOps side for the Fluent 2 website.
+This tool would have to support their design system library, which had a React version published.
+Extensibility was also important i.e. they wanted to be able to add CMS or DocSearch e.g. Algolia
+later on if they wanted to. Lastly, they wanted to be able to support legacy Storybook iframes to
+share code examples but be able to migrate off them eventually.
+
+## Solution and Implementation
+
+Astro ships zero Javascript by default, making it perfectly lightweight and performant for Microsoft.
```
Comment (ElianCodes) [inline src/content/caseStudies/microsoft/index.mdx:70]:
```suggestion
Astro ships zero JavaScript by default, making it perfectly lightweight and performant for Microsoft.
```

**Review by ElianCodes — COMMENTED** [review]
Just a small Typo

**Pull request merged**